gio-macros/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+extern crate proc_macro;
+
+mod action_map;
+
+use proc_macro::TokenStream;
+use syn::{parse_macro_input, ItemImpl};
+
+/// Macro for creating [`gio::Action`]s and registering them in a given [`gio::ActionMap`].
+/// It generates a method `register_actions` with a following signature:
+///
+/// ```rust
+/// # struct X; impl X {
+/// fn register_actions<AM: glib::object::IsA<gio::ActionMap>>(&self, map: &AM)
+/// # {} }
+/// ```
+///
+/// # Example
+/// ```rust
+/// use gio::prelude::*;
+///
+/// #[derive(glib::Downgrade)]
+/// pub struct MyApplication(gio::Application);
+///
+/// impl MyApplication {
+///     pub fn new() -> Self {
+///         let app = Self(gio::Application::new(
+///             Some("com.example.MyApplication"),
+///             gio::ApplicationFlags::FLAGS_NONE,
+///         ));
+///         app.register_actions(&app.0);
+///         app
+///     }
+/// }
+///
+/// #[gio::actions]
+/// impl MyApplication {
+///     fn action1(&self) {
+///         // handle "action1"
+///     }
+///
+///     // Explicitely specify action name
+///     #[action(name = "action_second")]
+///     fn action2(&self) {
+///         // handle "action_second"
+///     }
+///
+///     // Action with a parameter
+///     #[action(parameter_type = "s")]
+///     fn action3(&self, param: String) {
```

Comment:
What if the parameter type in the attribute and the function parameter don't match? Can we infer one from the other to not require both to be present? You can get the GVariant type from a Rust type in any case at runtime.

Reply:
It types doesn't match it will be a runtime error (`g_critical`). I am not sure, it is possible to derive the type without bloating the code (tuples of arbitrary length could be a PITA).

Reply:
The traits in `glib::variant` could be used for this. They already handle exactly these things.

Reply:
Oh, I overlooked that `StaticVarianType` exists. So, it seems I have something to do this evening :)

Reply:
This is done. And `parameter_type` parameter is removed.